protocol/contracts/trucoV1/TrucoMatchFactory.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.16;
+
+// Imports
+import '@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol';
+import '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import './interfaces/IERC3333.sol';
+import '../TrucoMatch.sol';
+
+contract TrucoMatchFactory is OwnableUpgradeable {
+    // State variables
+    TrucoMatch[] public matches;
+    IERC3333 internal trucoEngine;
+    IERC20 internal truCoin;
+    uint256 public minBet;
+
+    // Events
+    event TrucoMatchCreated(
+        address indexed match_address,
+        address indexed player1,
+        uint256 bet
+    );
+
+    // Initialize
+    function initialize(
+        IERC3333 _trucoEngine,
+        IERC20 _truCoin,
+        uint256 _minBet
+    ) public initializer {
+        trucoEngine = _trucoEngine;
+        truCoin = _truCoin;
+        minBet = _minBet;
+    }
+
+    // New Match
+    function newMatch(uint256 _bet) public returns (address) {
```

Comment:
I should return `TrucoMatch` instead

Reply:
My case for this change is that the data representation of a contract is identical to that of the address type and this type is also used in the ABI.

From the official documentation:
![Uploading image.png…]()